src/MLJCatBoostInterface.jl

```diff
@@ -0,0 +1,193 @@
+
+    cat_features = get_dtype_feature_ix(table_input, Multiclass) .- 1 # convert to 0 based indexing
+    text_features = get_dtype_feature_ix(table_input, MMI.Textual) .- 1 # convert to 0 based indexing
+    data_pool = Pool(table_input; label=numpy.array(Array(y)), cat_features, text_features)
+
+    return data_pool
+end
+
+function prepare_input(X)
+    table_input = Tables.columntable(X)
+    columns = Tables.columnnames(table_input)
+
+    order_factor_cols = columns[get_dtype_feature_ix(table_input, OrderedFactor)]
+    new_columns = NamedTuple{order_factor_cols}([MMI.int(table_input[col])
+                                                 for col in order_factor_cols])
+    table_input = (; drop_cols(table_input, order_factor_cols)..., new_columns...)
+
+    cat_features = get_dtype_feature_ix(table_input, Multiclass) .- 1 # convert to 0 based indexing
+    text_features = get_dtype_feature_ix(table_input, MMI.Textual) .- 1 # convert to 0 based indexing
+
+    X_pool = Pool(table_input; cat_features, text_features)
+
+    return X_pool
+end
+
+#####
+##### CatBoost Models
+#####
+
+include("mlj_catboostclassifier.jl")
+include("mlj_catboostregressor.jl")
+
+const CatBoostModels = Union{CatBoostClassifier,CatBoostRegressor}
+
+#####
+##### Reformat data for MLJ
+#####
+
+function MMI.reformat(::CatBoostRegressor, X, y)
+    data_pool = prepare_input(X, y)
+    return (data_pool,)
+end
+
+function MMI.reformat(::CatBoostClassifier, X, y)
+    data_pool = prepare_input(X, y)
+    return (data_pool, first(y))
+end
+
+function MMI.reformat(::CatBoostModels, X)
+    X_pool = prepare_input(X)
+    return (X_pool,)
+end
+
+function MMI.selectrows(::CatBoostModels, I, data_pool)
+    py_I = numpy.array(numpy.array(I .- 1))
+    return (data_pool.slice(py_I),)
+end
+
+function MMI.selectrows(::CatBoostModels, I::Colon, data_pool)
+    return (data_pool,)
+end
+
+function MMI.selectrows(::CatBoostClassifier, I, data_pool, y_first)
+    py_I = numpy.array(numpy.array(I .- 1))
+    return (data_pool.slice(py_I), y_first)
+end
+
+function MMI.selectrows(::CatBoostClassifier, I::Colon, data_pool, y_first)
+    return (data_pool, y_first)
+end
+
+#####
+##### Additional MLJ functionality 
+#####
+
+function MMI.update(mlj_model::CatBoostModels, verbosity::Integer, fitresult, cache,
+                    data_pool)
+    if mlj_model.iterations > cache.mlj_model.iterations &&
+       MMI.is_same_except(mlj_model, cache.mlj_model, :iterations)
+        iterations = mlj_model.iterations - cache.mlj_model.iterations
+        new_model = model_init(mlj_model; verbose, iterations)
+        new_model.fit(data_pool; init_model=fitresult)
+        report = (feature_importances=feature_importance(new_model),)
+        cache = (; mlj_model=mlj_model)
+    else
+        new_model, cache, report = fit(mlj_model, verbosity, data_pool)
+    end
+
+    return new_model, cache, report
+end
+
```

Comment:
Sorry, one last thing would be nice to include to save users having to provide it in MLJ's `IteratedModel` wrapper:

```suggestion
MMI.iteration_parameter(::Type{<:CatBoostModels}) = :iterations
```

Reply:
Done!